Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -359,72 +353,124 @@ - (void)configureViews
         }
     }
 
-    OWSMessageFooterView *footerView = self.footerView;
-    [footerView configureWithConversationViewItem:self.viewItem];
-    if (textStackView) {
-        // Display footer below text.
-        [textStackView addArrangedSubview:self.footerView];
-        [self.footerView setHasShadows:NO viewItem:self.viewItem];
-    } else if (bodyMediaView) {
+    if (self.viewItem.shouldHideFooter) {
+        // Do nothing.
+    } else if (shouldFooterOverlayMedia) {
+        OWSAssert(bodyMediaView);
         // Display footer over media.
-        [bodyMediaView addSubview:footerView];
+        [self.footerView configureWithConversationViewItem:self.viewItem hasShadows:YES];
+        [bodyMediaView addSubview:self.footerView];
 
         bodyMediaView.layoutMargins = UIEdgeInsetsZero;
         [self.viewConstraints addObjectsFromArray:@[
-            [footerView autoPinLeadingToSuperviewMarginWithInset:self.conversationStyle.textInsetHorizontal],
-            [footerView autoPinTrailingToSuperviewMarginWithInset:self.conversationStyle.textInsetHorizontal],
-            [footerView autoPinBottomToSuperviewMarginWithInset:self.conversationStyle.textInsetBottom],
+            [self.footerView autoPinLeadingToSuperviewMarginWithInset:self.conversationStyle.textInsetHorizontal],
+            [self.footerView autoPinTrailingToSuperviewMarginWithInset:self.conversationStyle.textInsetHorizontal],
+            [self.footerView autoPinBottomToSuperviewMarginWithInset:self.conversationStyle.textInsetBottom],
         ]];
-        [self.footerView setHasShadows:YES viewItem:self.viewItem];
     } else {
-        OWSFail(@"%@ could not display footer.", self.logTag);
+        // Display footer at bottom of message bubble.
+        [self.footerView configureWithConversationViewItem:self.viewItem hasShadows:NO];
+        [textViews addObject:self.footerView];
     }
 
-    if (textStackView) {
-        CGSize bubbleSize = [self measureSize];
-        [NSLayoutConstraint autoSetPriority:UILayoutPriorityRequired
-                             forConstraints:^{
-                                 [self.viewConstraints addObjectsFromArray:@[
-                                     [self autoSetDimension:ALDimensionWidth toSize:bubbleSize.width],
-                                 ]];
-                             }];
+    [self insertTextViewsIntoStackViewIfNecessary:textViews];
+
+    CGSize bubbleSize = [self measureSize];
+    [NSLayoutConstraint autoSetPriority:UILayoutPriorityRequired
+                         forConstraints:^{
+                             [self.viewConstraints addObjectsFromArray:@[
+                                 [self autoSetDimension:ALDimensionWidth toSize:bubbleSize.width],
+                             ]];
+                         }];
+
+    [self updateBubbleColorWithBodyMediaView:bodyMediaView];
+}
+
+- (void)updateBubbleColorWithBodyMediaView:(nullable UIView *)bodyMediaView
+{
+    BOOL hasOnlyBodyMediaView = NO;
```

Comment:
TIOLI naming nit: If found the current name confusing. Would something like `isMediaWithNoCaption` capture the same concept?

Reply:
I've reworked this in a subsequent PR - let's discuss there.